packages/babel-generator/src/printer.ts

```diff
@@ -1021,10 +1027,10 @@ class Printer {
               commentStartLine - (i === 0 ? nodeStartLine : lastLine);
             lastLine = commentEndLine;
 
-            this.newline(offset);
+            if (!_noLineTerminator) this.newline(offset);
             this._printComment(comment, COMMENT_SKIP_NEWLINE.SKIP_ALL);
 
-            if (i + 1 === len) {
+            if (!_noLineTerminator && i + 1 === len) {
               this.newline(Math.min(1, nodeEndLine - lastLine)); // TODO: Improve here when inner comments processing is stronger
               lastLine = nodeEndLine;
             }
```

Comment:
`lastLine` is only partially set, which seems a little dangerous, since this might be used elsewhere.
I prefer to not always set it or always set it.